libr/io/section.c

```diff
@@ -57,7 +57,10 @@ static RIOSection *findMatching (RIO *io, ut64 paddr, ut64 vaddr, ut64 size, ut6
 R_API RIOSection *r_io_section_add(RIO *io, ut64 offset, ut64 vaddr, ut64 size, ut64 vsize, int rwx, const char *name, ut32 bin_id, int fd) {
 	int update = 0;
 	RIOSection *s;
-	if (size == 0 || size > 0xf0000000) {
+	ut64 curr = lseek (fd, 0, SEEK_CUR);
```

Comment:
don't do this using lseek outside of an io-plugin is in general a bad idea. Here, in this case it is a very bad idea, which can make r_io explode, because not every fd must be a real filedescriptor. Somtimes a fd is a PID or a pointer to somewhere on the heap. So pls use the RIO-Api